# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1376397044 -7200
# Node ID 68e55e87d50d272b31ceb8f1e0c75f55d94f361f
# Parent  f6f838af9c67e75381758d8b6e4630de0ea43aaa
x86/watchdog: Tweak implementation given new common code

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r f6f838af9c67 -r 68e55e87d50d xen/arch/x86/nmi.c
--- a/xen/arch/x86/nmi.c
+++ b/xen/arch/x86/nmi.c
@@ -435,8 +435,7 @@ void nmi_watchdog_tick(struct cpu_user_r
 {
     unsigned int sum = this_cpu(nmi_timer_ticks);
 
-    if ( (this_cpu(last_irq_sums) == sum) &&
-         !atomic_read(&watchdog_disable_count) )
+    if ( (this_cpu(last_irq_sums) == sum) && watchdog_enabled() )
     {
         /*
          * Ayiee, looks like this CPU is stuck ... wait for the timeout
